Client/Ecosia/Extensions/LegacyThemeManager+Ecosia.swift

```diff
@@ -3,19 +3,44 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
 import UIKit
+import Common
 
 extension LegacyThemeManager {
-    // Ecosia: theme changing on a central place
+    /// Updates the theme based on the changes in the trait collection.
+    ///
+    /// - Parameters:
+    ///   - from: The previous trait collection. This is optional as it may be nil when there's no previous state to compare to.
+    ///   - to: The new trait collection after the change. This is used to determine the current theme settings.
+    ///   - forceDark: A boolean indicating whether the dark theme should be forced on regardless of the system theme. The default value is `false`.
     func themeChanged(from: UITraitCollection?, to: UITraitCollection, forceDark: Bool = false) {
+        // Determine if the color appearance has changed between the previous and new trait collections.
         let colorHasChanged = to.hasDifferentColorAppearance(comparedTo: from)
 
-
-        // Do not change theme if it was dark before night mode already
+        // Return early and do not change the theme if the color hasn't changed, dark mode is forced,
+        // or the system theme is already in effect.
         guard colorHasChanged,
               !forceDark,
               systemThemeIsOn else { return }
 
+        // Update the current theme based on the new user interface style.
         let userInterfaceStyle = to.userInterfaceStyle
         current = userInterfaceStyle == .dark ? LegacyDarkTheme() : LegacyNormalTheme()
     }
 }
+
+extension LegacyThemeManager {
+    
+    /// Updates the current theme based on the system theme type provided by the `EcosiaThemeManager`.
+    ///
+    /// This method retrieves the `EcosiaThemeManager` from the shared `AppContainer` and updates
+    /// the theme accordingly. It checks whether the system theme type is dark and sets the
+    /// `current` theme to `LegacyDarkTheme` or `LegacyNormalTheme` based on that.
+    func updateBasedOnCurrentSystemThemeType() {
+        // Safely unwrap the ThemeManager and cast it to EcosiaThemeManager. If this fails, return early.
+        guard let themeManager: ThemeManager = AppContainer.shared.resolve(),
+              let ecosiaThemeManager = themeManager as? EcosiaThemeManager else { return }
+        
+        // Update the current theme based on the system theme type.
+        LegacyThemeManager.instance.current = ecosiaThemeManager.getSystemThemeType() == .dark ? LegacyDarkTheme() : LegacyNormalTheme()
```

Comment:
I guess we only use `ecosiaThemeManager` to get the system type, can't we get it directly and avoid resolving the theme here?

```suggestion
        LegacyThemeManager.instance.current = UIScreen.main.traitCollection.userInterfaceStyle == .dark ? LegacyDarkTheme() : LegacyNormalTheme()
```

Reply:
For sure we can 👍 - the only reason for getting it from the `ecosiaThemeManager.getSystemThemeType()` is to make an "implicit" relation between the `EcosiaThemeManager` (latest) with our LegacyTheme.

Reply:
No strong opinions, just felt like a good thing getting rid of the theme manage resolution as that can crash things like https://github.com/ecosia/ios-browser/pull/771#discussion_r1742223716, though that has been fully fixed.

Maybe an alternative would be making `getSystemThemeType` static then so that you can directly use it without having to instantiate `EcosiaThemeManager`?